// @ng-toolkit/pwa/fesm5/ng-toolkit-pwa.js
import { Subject } from 'rxjs';
import { NgModule } from '@angular/core';
import { SwPush, SwUpdate } from '@angular/service-worker';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var SwUpdateServerMock = /** @class */ (function () {
    function SwUpdateServerMock() {
        this.available = new Subject();
        this.activated = new Subject();
        this.isEnabled = false;
    }
    /**
     * @return {?}
     */
    SwUpdateServerMock.prototype.checkForUpdate = /**
     * @return {?}
     */
    function () {
        return new Promise(function (resolve) { return resolve(); });
    };
    /**
     * @return {?}
     */
    SwUpdateServerMock.prototype.activateUpdate = /**
     * @return {?}
     */
    function () {
        return new Promise(function (resolve) { return resolve(); });
    };
    return SwUpdateServerMock;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var SwPushServerMock = /** @class */ (function () {
    function SwPushServerMock() {
    }
    /**
     * @param {?} options
     * @return {?}
     */
    SwPushServerMock.prototype.requestSubscription = /**
     * @param {?} options
     * @return {?}
     */
    function (options) {
        return new Promise(function (resolve) { return resolve(); });
    };
    /**
     * @return {?}
     */
    SwPushServerMock.prototype.unsubscribe = /**
     * @return {?}
     */
    function () {
        return new Promise(function (resolve) { return resolve(); });
    };
    return SwPushServerMock;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var NgtPwaMockModule = /** @class */ (function () {
    function NgtPwaMockModule() {
    }
    NgtPwaMockModule.decorators = [
        { type: NgModule, args: [{
                    providers: [
                        { provide: SwUpdate, useClass: SwUpdateServerMock },
                        { provide: SwPush, useClass: SwPushServerMock }
                    ]
                },] },
    ];
    return NgtPwaMockModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { NgtPwaMockModule, SwPushServerMock, SwUpdateServerMock };
